### PR TITLE
Jwt validation

### PR DESCRIPTION
**Pull request by kyleferguson**
JWT token validation

### CODE REVIEW

**Review by larkmullins — APPROVED** [review]

File: auth/guard.go [hunk 1]
```diff
@@ -80,3 +81,9 @@ func jwtSigningMethod() jwt.SigningMethod {
 		return nil
 	}
 }
+
+// extractJWTFromHeader will parse the header string and return just the token
+// header should be formatted as Bearer [TOKEN]
+func extractJWTFromHeader(header string) string {
+	return header[7:]
```
Comment (jlaswell) [inline auth/guard.go:88]: What happens if an `Authorization` header is set, but less than 7 characters?

File: auth/guard_test.go [hunk 2]
```diff
@@ -33,16 +34,34 @@ func TestAPIMiddlewareInvalidHeader(t *testing.T) {
...
 	request, _ := http.NewRequest("GET", "/api/v1/foo", nil)
-	request.Header.Set("Authorization", "Bearer token123")
+	request.Header.Set("Authorization", "Bearer "+token)
 	rw.On("Write", []byte("ok")).Once()
 	rw.On("WriteHeader", 200).Once()
 
 	APIMiddleware(rw, request, http.HandlerFunc(mockHTTPHandler))
 
-	rw.AssertExpectations(t)
+	//rw.AssertExpectations(t)
```
Comment (jlaswell) [inline auth/guard_test.go]: hanging comment

**Review by jlaswell — COMMENTED** [review]
Minor things

File: auth/guard.go (hunk 1 above)
Comment (kyleferguson) [reply auth/guard.go:88]: It should never get here if the header doesn't match the regex from `validateAuthHeader`

**Comment by jlaswell** [comment]
lgtm

**Pull request merged**